channels/applicationinsights-channel-js/src/Sender.ts

```diff
@@ -216,14 +216,82 @@ export class Sender extends BaseTelemetryPlugin implements IChannelControls {
                     _clearScheduledTimer();
 
                     try {
-                        return _self.triggerSend(isAsync, null, sendReason || SendRequestReason.ManualFlush);
+                        if (callBack) {
```

Comment:
Revert all of these changes.
- The root of the issue is that the call to _doSend in the _doSend never returns an IPromise it only ever returns null

If the call to _self.triggerSend doesn't return a promise then it should call the callback here

Reply:
The implementation should be simple, concise and avoid duplicate code and nested exception handling